### PR TITLE
Initial infrastructure

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: src/app/log.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/app/log.hpp:2]: Should we set 2024 year to all new files?
```suggestion
 * Copyright (C) 2024 Renesas Electronics Corporation.
```

File: src/utils/exception.hpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (mlohvynenko) [inline src/utils/exception.hpp:7]: Include guard is missing

File: src/app/log.hpp (hunk 1 above)
Comment (al1img) [reply src/app/log.hpp:2]: done

File: src/utils/exception.hpp (hunk 2 above)
Comment (al1img) [reply src/utils/exception.hpp:7]: done

**Review by mlohvynenko — APPROVED** [review]

File: src/utils/exception.hpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+#ifndef EXCEPTION_HPP_
+#define EXCEPTION_HPP_
```
Comment (mykola-kobets-epam) [inline src/utils/exception.hpp:9]: header files should be separated from source files in some way.
I see two options: 
1) make include / src folders for each subcomponent & make include directories as public properties for targets that depend on them
2) follow core_lib_cpp approach: have root/src and root/include dir.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET aos_iamanager_test)
+
+# ######################################################################################################################
+# Sources
+# ######################################################################################################################
+
+set(SOURCES ../../src/logger/logger.cpp src/exception_test.cpp)
```
Comment (mykola-kobets-epam) [inline tests/CMakeLists.txt:14]: I would better make test/CMakeLists.txt to include cmake files as sub_directories.

File: src/app/app.hpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (mlohvynenko) [inline src/app/app.hpp:7]: Include guard is missing

File: src/app/app.hpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Aos IAM application.
+ */
+class App : public Poco::Util::ServerApplication {
+public:
+    /**
+     * Constructs new IAM application.
+     */
+    App() { }
```
Comment (MykolaSolyanko) [inline src/app/app.hpp:20]: better to use the default keyword instead of explicitly setting an empty body

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

File: tests/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+# Includes
+# ######################################################################################################################
+
+include_directories(${CMAKE_SOURCE_DIR}/src)
+
+# ######################################################################################################################
+# Sources
+# ######################################################################################################################
+
+set(SOURCES utils/exception_test.cpp)
```
Comment (mykola-kobets-epam) [inline tests/CMakeLists.txt:24]: redundat

File: src/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+# ######################################################################################################################
+
+install(TARGETS ${TARGET} RUNTIME DESTINATION bin)
+
+# ######################################################################################################################
+# Modules
+# ######################################################################################################################
+
+add_subdirectory(app)
+add_subdirectory(logger)
```
Comment (mykola-kobets-epam) [inline src/CMakeLists.txt:49]: I think the ordering is not good:
modules should be imported before they are linked

File: src/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+# ######################################################################################################################
+# Target
+# ######################################################################################################################
+
+add_executable(${TARGET} ${SOURCES})
+
+# ######################################################################################################################
+# Labraries
+# ######################################################################################################################
```
Comment (mykola-kobets-epam) [inline src/CMakeLists.txt:34]: typo

File: src/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+# Copyright (C) 2024 EPAM Systems, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET aos_iamanager)
+
+# ######################################################################################################################
+# Defines
+# ######################################################################################################################
```
Comment (mykola-kobets-epam) [inline src/CMakeLists.txt:12]: empty section is not needed

File: tests/utils/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+# Target
+# ######################################################################################################################
+
+add_library(${TARGET} STATIC ${SOURCES})
+
+# ######################################################################################################################
+# Labraries
+# ######################################################################################################################
+
+target_link_libraries(${TARGET} PUBLIC aoscommon GTest::gtest)
```
Comment (mykola-kobets-epam) [inline tests/utils/CMakeLists.txt:30]: in theory it should be enough to link utils static library & aoscommon will be linked transitively

**Review by mykola-kobets-epam — APPROVED** [review]

File: tests/utils/CMakeLists.txt (hunk 11 above)
Comment (al1img) [reply tests/utils/CMakeLists.txt:30]: this is on theory. Practically, aos_common is not linked.

File: src/CMakeLists.txt (hunk 8 above)
Comment (al1img) [reply src/CMakeLists.txt]: Doesn't matter actually, but I'm ok to move them to the top.

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**